tensorzero-core/src/optimization/together_sft/mod.rs

```diff
@@ -688,21 +633,42 @@ impl Optimizer for TogetherSFTConfig {
             Some(0)
         };
 
-        // Build configurations using clean impl methods
-        let lr_scheduler = TogetherLRScheduler::from_config(
-            self.lr_scheduler_type.as_deref(),
-            self.lr_scheduler_min_lr_ratio,
-        )?;
+        // Build API configurations with defaults
+        let lr_scheduler = self
+            .lr_scheduler
+            .clone()
+            .unwrap_or(TogetherLRScheduler::Linear { min_lr_ratio: 0.0 });
 
-        let training_method = TogetherTrainingMethod::from_config(self.train_on_inputs.as_deref());
+        let training_method = match &self.training_method {
+            TogetherTrainingMethod::Sft { train_on_inputs } => TogetherTrainingMethod::Sft {
+                train_on_inputs: train_on_inputs.clone().or_else(|| Some("auto".to_string())),
+            },
+        };
 
-        let training_type = TogetherTrainingType::from_config(
-            self.training_type.as_deref(),
-            self.lora_r,
-            self.lora_alpha,
-            self.lora_dropout,
-            self.lora_trainable_modules.as_deref(),
-        )?;
+        let training_type = match &self.training_type {
+            Some(TogetherTrainingType::Full {}) => TogetherTrainingType::Full {},
+            Some(TogetherTrainingType::Lora {
+                lora_r,
+                lora_alpha,
+                lora_dropout,
+                lora_trainable_modules,
+            }) => TogetherTrainingType::Lora {
+                lora_r: lora_r.or(Some(8)),
+                lora_alpha: lora_alpha.or(Some(32)),
+                lora_dropout: lora_dropout.or(Some(0.0)),
+                lora_trainable_modules: Some(
+                    lora_trainable_modules
+                        .clone()
+                        .unwrap_or_else(|| "all-linear".to_string()),
+                ),
+            },
+            None => TogetherTrainingType::Lora {
```

Comment:
let's implement Default on TogetherTrainingType and then make this mandatory other places

Reply:
then you should be able to do #[serde(default)] and similar. In general, if we're gonna make default values for fields we should use the Default trait because then all code will know about it.